### PR TITLE
💄 👽 🌐 #520: hookup header API, tidy dashboard

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI
- Connect header to /users endpoint to get the user's name.
- Update Dashboard to show "consent incomplete" state by default.
- Update Dashboard to have an RSC wrapper with suspense boundary, for future API connection.

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/views/Dashboard/index.tsx [hunk 1]
```diff
@@ -17,69 +17,26 @@
...
-								size="large"
-								action="next"
-							>
-								{translate('dashboard', 'completeConsentForms')}
-							</LocalizedLink>
-						)}
-					</div>
-				</div>
-			</Card>
+			<Suspense
```
Comment (samrichca) [inline apps/consent-ui/src/components/views/Dashboard/index.tsx:33]: I moved the content of this component to a new component and added a suspense wrapper.

File: apps/consent-ui/src/components/views/Dashboard/notifications/DashboardNotification/index.tsx [hunk 2]
```diff
@@ -28,17 +28,16 @@ import { NotificationTranslations } from 'src/components/providers/NotificationP
...
 	currentLang,
 	notificationTranslations,
 }: {
+	consentInProgress?: boolean;
 	currentLang: ValidLanguage;
 	notificationTranslations: NotificationTranslations;
 }) => {
 	const { showNotification } = useNotification();
 
-	// STUB - get consent progress from API
```
Comment (samrichca) [inline apps/consent-ui/src/components/views/Dashboard/notifications/DashboardNotification/index.tsx:39]: I'm going to fetch this info in the Dashboard component. emailVerifiedParam still TBD

File: apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const Dashboard = async ({ currentLang }: { currentLang: ValidLanguage }) => {
+	const { translate } = getTranslation(currentLang);
+	const notificationTranslations = getNotificationTranslations(currentLang);
+
+	const { FEATURE_CONSENT_PDF_BUTTONS } = getAppConfig();
+
+	return (
+		<>
+			<DashboardNotification
+				consentInProgress
```
Comment (samrichca) [inline apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx:48]: the dashboard is currently in the incomplete/in-progress state.

File: apps/consent-ui/src/components/common/Header/handleGetUser.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { UserResponse } from 'types/consentApi';
+import { Result, SystemError, success, failure } from 'types/httpResponses';
+
+import { getUser } from 'src/services/api';
+import { auth } from 'src/app/auth';
+
+type UserFailureState = 'NO_SESSION' | SystemError;
```
Comment (anncatton) [inline apps/consent-ui/src/components/common/Header/handleGetUser.tsx:26]: we can get rid of the `NO_SESSION` status type - when we have proper error handling we'll be looking at the 401 response from consent-api.

Also smol thing, let's rename the error type to `GetUserFailureStatus`

File: apps/consent-ui/src/components/common/Header/Header.tsx [hunk 5]
```diff
@@ -111,8 +115,8 @@ const Header = ({ currentLang, textDict, session, languageToggleProps }: HeaderC
 					<div className={styles.userMenu}>
 						{/* Desktop */}
 						<div className={styles.desktopUserMenu}>
-							{session?.user ? (
-								<div>Hello, {session.user.preferredUsername}</div>
+							{userDisplayName ? (
+								<div>Hello, {userDisplayName}</div>
```
Comment (anncatton) [inline apps/consent-ui/src/components/common/Header/Header.tsx:119]: not a blocker for this pr, but can we throw in the icon from the mocks if it's handy?

File: apps/consent-ui/src/services/api/requests/getUser.ts [hunk 6]
```diff
@@ -17,26 +17,25 @@
...
 import consentApiFetch from '../axios/consentApiFetch';
+import catchAxiosError from '../axios/catchAxiosError';
 
-export const getUser = async (session: Session | null): Promise<User | undefined> => {
+/**
+ * Fetch user's data by their session data from consent API.
+ * @param {Session | null} session
+ * @returns HandledResponse<UserResponse>
+ */
+export const getUser = async (session: Session | null): HandledResponse<UserResponse> => {
```
Comment (anncatton) [inline apps/consent-ui/src/services/api/requests/getUser.ts:34]: 👍

File: apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+				<div className={styles['consents-img']}>
+					<Image src={ConsentsImage} alt="" priority />
+				</div>
+				<div className={styles.content}>
+					<h2>{translate('dashboard', 'reviewOhcrnConsents')}</h2>
+					<p>{translate('dashboard', 'reviewConsentsDescription')}</p>
+					{FEATURE_CONSENT_PDF_BUTTONS && <TempPdfButtonsWrapper currentLang={currentLang} />}
+					<div className={styles['button-container']}>
+						{consentStatus === 'incomplete' ? (
+							<LocalizedLink
```
Comment (anncatton) [inline apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx:62]: yay 🎉

File: apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx (hunk 3 above)
Comment (anncatton) [reply apps/consent-ui/src/components/views/Dashboard/Dashboard.tsx:48]: should we add a todo here just to call out this is hardcoded?

File: apps/consent-ui/src/components/common/Header/index.tsx [hunk 8]
```diff
@@ -34,14 +34,22 @@ const HeaderWrapper = async ({ currentLang }: { currentLang: ValidLanguage }) =>
...
-	const session = await auth();
+	// request user's name to display in the header
+	const userResult = await handleGetUser();
+	if (userResult && userResult.status !== 'SUCCESS') {
+		// TODO: Adding a log for testing purpose, but should implement proper error handling
+		console.error(userResult.status);
+	}
+
+	const { firstName, preferredName } = userResult?.data || {};
+	const userDisplayName = preferredName || firstName;
```
Comment (anncatton) [inline apps/consent-ui/src/components/common/Header/index.tsx:45]: 👍

**Review by anncatton — COMMENTED** [review]
looks good, just some minor comments/questions. Very happy to see the endless loading button gone!

**Review by anncatton — APPROVED** [review]
thanks for pepping up the styling! 🚀

**Pull request merged**